src/bson_mapper/bson_archiver.hpp

```diff
@@ -0,0 +1,1078 @@
+#ifndef CEREAL_ARCHIVES_BSON_HPP_
```

Comment:
Generally, you'll want to use macros with prefixes specific to your project.  In this case BSON_MAPPER_XXX would do the trick.